bpf/lib/policy.h

```diff
@@ -139,6 +139,26 @@ static inline int __inline__
 __policy_can_access(void *map, struct __sk_buff *skb, __u32 identity,
 		    __u16 dport, __u8 proto, int dir, bool is_fragment)
 {
+#ifdef ALLOW_ICMP_FRAG_NEEDED
+	// When ALLOW_ICMP_FRAG_NEEDED is defined we allow all packets
+	// of ICMP type 3 code 4 - Fragmentation Needed
+	if (proto == IPPROTO_ICMP) {
+		void *data, *data_end;
+		struct icmphdr icmphdr;
+		struct iphdr *ip4;
+		__u32 off = ((void *)ip4 - data) + ipv4_hdrlen(ip4);
```

Comment:
You probably wanted to calculate the offset after calling the `revalidate_data()` below?

Reply:
Sorry for this.
Can you tell me like where and how can I add tests for this?

Reply:
You could write a smoke test which would actually do a PMTU discovery. To begin with, check https://docs.cilium.io/en/latest/contributing/#end-to-end-testing-framework and existing tests at https://github.com/cilium/cilium/tree/master/test/k8sT or https://github.com/cilium/cilium/tree/master/test/runtime.